### PR TITLE
Spreadsheet+LibGUI: More fixes and improvements

### LINKED ISSUE #4167: Spreadsheet: Strange copy/paste and selection behaviour
There are several issues with the copying and pasting of cells.

I'm not comfortable with its current default behaviour of dragging copying a cell. When I click and drag in a spreadsheet then I am usually wanting to select a rectangular region of cells or a section of one row or column. Could we make this possible please?

If I drag and copy multiple cells, sometimes they get posted above and sometimes below the cursor.

After pasting some cells, they often don't appear until I change the cell selection.

I'd prefer selecting, copying and pasting of cells to work like LO Calc / Excel ideally.

### LINKED ISSUE #4168: Spreadsheet: DELETE doesn't work as expected
Enter some data in a cell
Click on the cell (once)
Hit the DELETE key
Click on another cell 

You'll notice the data remains in the supposedly DELETE'd cell.

### LINKED ISSUE #4170: Spreadsheet: Junk characters added to csv exports
In the attached screenshot we see what I input to spreadsheet and what it output. Spreadsheet has added a line with the letters A to P separated by commas, several commas after my input (on the second line) and then several additional lines of unneeded commas.

The correct output should've been simply

```
1,2
``` 

csv-export

### LINKED ISSUE #4171: Spreadsheet: Crash when importing CSV
I have tried loading a .csv file that only contains the line

```
1,2,3
```

into Spreadsheet but it crashes.

### PR DESCRIPTION
**Pull request by alimpfard**
Fixing a whole bunch of things
- [x] Fixes #4167 
    - [x] "proper" drag-to-select
    - [x] update on paste
- [x] Fixes #4168 
- [x] Fixes #4170 
    - [x] "infinite" scroll (rows)
    - [x] "infinite" scroll (columns)
- [x] Fixes #4171

### CODE REVIEW

File: Libraries/LibGUI/TableView.cpp [hunk 1]
```diff
@@ -177,11 +177,12 @@ void TableView::keydown_event(KeyEvent& event)
     if (event.is_accepted())
         return;
 
-    if (is_editable() && edit_triggers() & EditTrigger::AnyKeyPressed && !event.text().is_empty()) {
+    auto is_delete = event.key() == Key_Delete || event.key() == Key_Backspace;
+    if (is_editable() && edit_triggers() & EditTrigger::AnyKeyPressed && (event.code_point() != 0 || is_delete)) {
```
Comment (alimpfard) [inline Libraries/LibGUI/TableView.cpp:181]: Someone explain to me why `event.code_point() == '.'` when `event.key() == Key_Delete`.

File: Libraries/LibGUI/Model.cpp [hunk 2]
```diff
@@ -57,10 +59,6 @@ void Model::did_update(unsigned flags)
 {
     for (auto* client : m_clients)
         client->model_did_update(flags);
-
-    for_each_view([&](auto& view) {
-        view.did_update_model(flags);
-    });
```
Comment (awesomekling) [inline Libraries/LibGUI/Model.cpp:63]: I just want to say that I love this change.

File: Libraries/LibGUI/TableView.cpp (hunk 1 above)
Comment (awesomekling) [reply Libraries/LibGUI/TableView.cpp:181]: I don't know. That's strange. The code point is derived from the current character map fwiw.

File: Applications/Spreadsheet/SpreadsheetWidget.cpp [hunk 3]
```diff
@@ -175,6 +175,11 @@ void SpreadsheetWidget::load(const StringView& filename)
         GUI::MessageBox::show_error(window(), result.error());
         return;
     }
+
+    m_tab_widget->on_change = nullptr;
+    m_cell_value_editor->on_change = nullptr;
+    m_current_cell_label->set_text("");
+    m_should_change_selected_cells = false;
```
Comment (awesomekling) [inline Applications/Spreadsheet/SpreadsheetWidget.cpp:182]: This sort of thing is always gnarly and it's nice to solve with e.g `WeakPtr` when possible.
I don't mind doing it this way, just lamenting that it's a bit of a bitch :)

File: Libraries/LibGfx/Point.h [hunk 4]
```diff
@@ -208,6 +208,11 @@ class Point {
         return sqrtf(powf(m_x - other.m_x, 2.0f) + powf(m_y - other.m_y, 2.0f));
     }
 
+    Point absolute_relative_distance_to(const Point& other) const
```
Comment (awesomekling) [inline Libraries/LibGfx/Point.h:211]: This name is weird, given that "absolute" and "relative" are antonyms.

Comment (alimpfard) [reply Libraries/LibGfx/Point.h:211]: ...I didn't even think how odd that sounds
My name-fu is not that strong, it seems. Oops.

File: Applications/Spreadsheet/SpreadsheetWidget.cpp (hunk 3 above)
Comment (alimpfard) [reply Applications/Spreadsheet/SpreadsheetWidget.cpp:182]: Hmmm...weak pointer to a function?

File: Libraries/LibGUI/Model.cpp (hunk 2 above)
Comment (alimpfard) [reply Libraries/LibGUI/Model.cpp:63]: Me too!

**Pull request merged**